runtime/bindings/python/CMakeLists.txt

```diff
@@ -4,9 +4,23 @@
 # See https://llvm.org/LICENSE.txt for license information.
 # SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
 
-set(NUMPY_DEPS "")
-set(PYBIND_COPTS "-fexceptions")
-set(PYBIND_EXTENSION_COPTS "-fvisibility=hidden")
+if(NOT nanobind_FOUND)
+  # nanobind requires Python >= 3.8.
+  find_package(Python 3.8 COMPONENTS Interpreter Development.Module NumPy REQUIRED)
```

Comment:
Should we check for Python >= 3.8 earlier, in the root CMakeLists?
https://github.com/openxla/iree/blob/59d0988c2b0a2640cd38109f2737f9474c5f0c11/CMakeLists.txt#L608-L623

(otherwise maybe the root could find 3.7 and that would get out of sync with what is used/needed here?)

Reply:
Added.